package com.kh.chap03_throws.controller;

import java.io.IOException;

public class ThrowsController {
	
	public void method1 () {
		System.out.println("method1() 호출");
		try {
			method2();			
		} catch (IOException e) {
			System.out.println("예외 발생");
			e.printStackTrace();
		}
	}
	
	public void method2 () throws IOException {
		System.out.println("method2() 호출");
		method3();
	}
	
	public void method3 () throws IOException {
		System.out.println("method3() 호출");
		method4();
	}
	
	public void method4 () {
		System.out.println("method4() 호출");
		
		// throw : 강제로 Exception 발생시키는 키워드
		// throws : 발생된 Exception을 호출부로 전가하는 키워드
		throw new IOException();
	}
	
	
	

}
